### PR TITLE
Compile fails with Arduino SAMD board package	1.8.10 and newer

### PR DESCRIPTION
**Pull request by gdsports**
Proposed fix for https://github.com/NicoHood/HID/issues/242

Compiles on the following SAMD board packages: Arduino, Adafruit, and Seeeduino.

### CODE REVIEW

File: src/HID-Settings.h [hunk 1]
```diff
@@ -111,12 +111,17 @@ THE SOFTWARE.
 
 #define ATTRIBUTE_PACKED  __attribute__((packed, aligned(1)))
 
+#define USB_EP_SIZE                 EPX_SIZE
+#define EP_TYPE_INTERRUPT_IN        USB_ENDPOINT_TYPE_INTERRUPT | USB_ENDPOINT_IN(0);
+#define EP_TYPE_INTERRUPT_OUT       USB_ENDPOINT_TYPE_INTERRUPT | USB_ENDPOINT_OUT(0);
+
+#if defined(ARDUINO_API_VERSION)
+#include "api/PluggableUSB.h"
+#define EPTYPE_DESCRIPTOR_SIZE		unsigned int
```
Comment (NicoHood) [inline src/HID-Settings.h:120]: why is it unsigned int here? If so, please use uint16_t

File: src/port/samd.cpp [hunk 2]
```diff
@@ -2,12 +2,14 @@
 
 #ifdef ARDUINO_ARCH_SAMD
 
+#if !defined(ARDUINO_API_VERSION)
```
Comment (NicoHood) [inline src/port/samd.cpp:5]: Why are those not required anymore? Are they implemented by the api?

File: src/HID-Settings.h (hunk 1 above)
Comment (gdsports) [reply src/HID-Settings.h:120]: These changes are based on similar changes the Arduino folks made to their MIDIUSB library having to do with ARDUINO_API. I suspect the idea is to have the same USB API for AVR and SAMD. unsigned int will be 16 bits for AVR and 32 bits for SAMD. I have no direct information on the purpose for the ARDUINO_API changes.

https://github.com/arduino-libraries/MIDIUSB/commit/80b6fac501702a268d7f9ab9702333c5d059e059

File: src/port/samd.cpp (hunk 2 above)
Comment (gdsports) [reply src/port/samd.cpp:5]: Yes, the SAMD board release now has USBAPI.h which include routines for USB_SendControl and others.

https://github.com/arduino/ArduinoCore-API/blob/master/api/USBAPI.h

As far as I can tell, the latest SAMD board package releases no longer match github.com/arduino/ArduinoCore-SAMD. It appears the files from ArduinoCore-API are included in the SAMD board packages during the release build procedure.

File: src/HID-Settings.h (hunk 1 above)
Comment (NicoHood) [reply src/HID-Settings.h:120]: Alright, that makes sense. Was not aware that its 32 bit on SAMD

File: src/port/samd.cpp (hunk 2 above)
Comment (NicoHood) [reply src/port/samd.cpp:5]: Would you mind finding this out for avr? It would be best to have a solution that works with every board, of course ;-)

Comment (NicoHood) [reply src/port/samd.cpp:5]: Any chance to test the AVR code? I'd be happy to merge then ;-)

**Comment by NicoHood** [comment]
@SukkoPera Maybe you can give this a regression test on AVR and see if it still works? That'd be nice, so we can include it in the next release along with your keylayout fixes.

**Comment by SukkoPera** [comment]
Everything compiles fine for ATmega32U4. I did a quick 5 minute test with SingleGamepad and BootKeyboard and everything seems to be fine. Otherwise you wouldn't be reading this :).

**Pull request merged**